dts: bindings: display: Change the property names in the overlay

rruuaanng opened the pull request:
Unify property names in bindings and overrides, using hyphens (-) instead of underscores (_) as separators.

kartben reviewed (changes requested):
this is missing a entry in migration guide, you can't just rename properties without documenting it, this will break downstream users. 
also commit message should be "drivers: display: ..." imo, the "root" change is in the display bindings, not in shields

rruuaanng commented:
> this is missing a entry in migration guide, you can't just rename properties without documenting it, this will break downstream users. 
> 
> also commit message should be "drivers: display: ..." imo, the "root" change is in the display bindings, not in shields

This PR is just a sub-PR of #83352.

rruuaanng commented:
> this is missing a entry in migration guide, you can't just rename properties without documenting it, this will break downstream users. 
> 
> also commit message should be "drivers: display: ..." imo, the "root" change is in the display bindings, not in shields

This PR is just a sub-PR of #83352.

rruuaanng commented:
Because the commit was too large, I needed to split it. Additionally, I will modify the commit messages later.

rruuaanng committed:
dts: bindings: display: Change the property names in the overlay

Unify property names in bindings and overlay, using
hyphens (-) instead of underscores (_) as separators.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
doc: Add a new migration guide entry to display

Rename the pclk_pol and data_cmd-gpios propertys
to pclk-pol and data-cmd-gpios in the devicetree
and bindings of the display subsystem.

Signed-off-by: James Roy <rruuaanng@outlook.com>
